factory/usr/lib/the-modeenv

```diff
@@ -5,10 +5,12 @@ cp /sysroot/run/image.fstab /run/
 # TODO:UC20: move this logic into snap-bootstrap to be incorporated with 
 # initramfs-mounts
 
+mode="$(/usr/libexec/core/get-mode mode)" || mode="$(/usr/libexec/core/get-arg snapd_recovery_mode)" || mode="unknown"
```

Comment:
I like this.
I cannot find `/usr/libexec/core/get-mode` though. Have you forgot to include it in the commit?